src/common/ini_handler.c

```diff
@@ -18,68 +17,89 @@ static int ini_handler_func(void *user, const char *section, const char *name, c
 
     ETH_config_t *tmp_config = (ETH_config_t *)user;
 
-    if (ini_string_match(section, "lan_ip4", name, "type")) {
-        if (strncasecmp(value, "DHCP", 4) == 0) {
+    if (ini_string_match(section, "eth::ipv4", name, "type")) {
+        if ((strcmp(value, "DHCP") == 0) || (strcmp(value, "dhcp") == 0)) {
```

Comment:
ideally `strncasecmp(value, "DHCP", 4)` just like it was before

Reply:
done